src/components/pages/labs/categories/categories.jsx

```diff
@@ -1,33 +1,52 @@
 import clsx from 'clsx';
 import { navigate } from 'gatsby';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useRef, useEffect } from 'react';
 
 const Categories = ({ categories, currentCategory, className }) => {
+  const categoryRefs = useRef([]);
+
   const handleCategoryClick = (slug) => (event) => {
     event.preventDefault();
-    navigate(slug, {
-      state: { preventScroll: true },
-    });
+    navigate(slug);
   };
 
+  useEffect(() => {
+    const activeIndex = categories.findIndex(({ name }) => name === currentCategory);
+    if (activeIndex !== -1 && categoryRefs.current[activeIndex]) {
+      categoryRefs.current[activeIndex].scrollIntoView({
+        behavior: 'smooth',
+        block: 'nearest',
+      });
+    }
+  }, [categories, currentCategory]);
+
```

Comment:
This function doesn't work for me either. I'd suggest removing it for now, and trying to test the API https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#shouldUpdateScroll